arrow-json/src/writer/mod.rs

```diff
@@ -1938,18 +1968,116 @@ mod tests {
         let field = Arc::new(Field::new("decimal", array.data_type().clone(), true));
         let schema = Schema::new(vec![field]);
         let batch = RecordBatch::try_new(Arc::new(schema), vec![Arc::new(array)]).unwrap();
+        assert_decimal_outputs(
+            &batch,
+            r#"{"decimal":12.34}
+{}
+{"decimal":56.78}
+"#,
+            r#"{"decimal":"12.34"}
+{}
+{"decimal":"56.78"}
+"#,
+        );
+    }
 
-        let mut buf = Vec::new();
-        {
-            let mut writer = LineDelimitedWriter::new(&mut buf);
-            writer.write_batches(&[&batch]).unwrap();
+    #[test]
+    fn test_decimal_list_encoder_as_string() {
```

Comment:
It seems `*_as_string` doesn't make any sense, because the both outputs are asserted anyway. And let's rename the test to reflect the target type.
```suggestion
    fn test_decimal128_list_encoder() {
```
If you agree, let's apply this to the rest:
- `test_decimal_dictionary_encoder_as_string` -> `test_decimal128_dictionary_encoder`
- `test_decimal256_dictionary_encoder_as_string` -> `test_decimal256_dictionary_encoder`

Reply:
Done